lib/spack/spack/cmd/mirror.py

```diff
@@ -658,19 +666,31 @@ def _specs_and_action(args):
     return mirror_specs, mirror_fn
 
 
-def create_mirror_for_all_specs(mirror_specs, path, skip_unstable_versions):
+def create_mirror_for_one_spec(candidate, mirror_cache, mirror_stats):
+    pkg_cls = spack.repo.PATH.get_pkg_class(candidate.name)
+    pkg_obj = pkg_cls(spack.spec.Spec(candidate))
+    spack.mirrors.utils.create_mirror_from_package_object(pkg_obj, mirror_cache, mirror_stats)
```

Comment:
You're still mutating `mirror_stats` in a subprocess. The function should return what it added / what errored. Do the reduction in the main process.